src/gsTensor/gsTensorFitting.h

```diff
@@ -0,0 +1,355 @@
+		* @param[in] n_p	number of data points in other direction
+		* @param[in] Q	3x(m*n_p)-matrix of data points, each col is a point
+		* @param[out] R	((m-2)*n_px3)-matrix containing righthandside needed in least squares, like (9.67), p.412, in nurbs book
+		* @param[in] stp	either 0 or 1 for first resp second direction
+		* @param[out] is_bad	indicate if system in fitting-step will be solvable, = true if not solvable
+		* @warning	only stp=0 or stp=1 allowed
+		* @warning	if is_bad=true, then no values for R or B will be computed
+		*/ 
+		void make_basis_matrix(const gsKnotVector<T>& kv, const std::vector<T>& u_k, const index_t m, const size_t n, const index_t p, gsMatrix<T>& B, const size_t n_p, const gsMatrix<T>& Q, gsMatrix<T>& R, const int stp, bool& is_bad); 
+		
+		/** does the fitting step by calling make_basis_matrix and solving the least squares system
+		* @param[in] n_p	number of curves in 'other' direction
+		* @param[in] n_q	number of desired control points along 'this' direction
+		* @param[in] m_q	number of given data points along 'this' direction
+		* @param[in] kv_v	knot vector used to construct basis
+		* @param[in] q	degree of kv_v
+		* @param[in] v_k	vector containing parameter values, size(v_k) = m_q
+		* @param[in] P_p	3x(n_p*m_q)-matrix containing approximation points - each col is a point
+		* @param[out] P_points	3x(n_p*n_q)-matrix containing control points after fitting
+		* @param[in] stp	indicates if first (stp=0) or second (stp=1) fitting step is made
+		* @param[out] is_bad	= true if system is not invertible -> in this case, no points are computed
+		* @warning	no check if stp==1 or stp==2 only
+		*/
+		void fitting_step(const size_t n_p, const size_t n_q, const index_t m_q, const gsKnotVector<T>& kv_v, const index_t q, const std::vector<T>& v_k, const gsMatrix<T>& P_p, gsMatrix<T>& P_points, const int stp, bool& is_bad); 
+		
+		//----------------------------------------------------------------------
+		//measuring the distance between approximation and desired points
+		/** checks if any of convergence criteria for point projection is satisfied
+		* cf. nurbs book, p.233, (6.8)
+		* @param[in] R	distance between evaluated surface and desired point 
+		* @param[in] k	contains f(u,v) and g(u,v), see p. 232, (6.5)
+		* @param[in] Su	first derivatives in u direction
+		* @param[in] Sv	first derivatives in v direction
+		* @param[in] eps1	epsilon for point coincidence
+		* @param[in] eps2	epsilon for zero cosine check
+		* @return bool-variable which is true if either point coincidence or zero cosine is satisfied
+		*/
+		bool check_conv_crit(const gsMatrix<T>& R, const std::vector<T>& k, const std::vector<T>& Su, const std::vector<T>& Sv, const T& eps1, const T& eps2); 
+		
+		/** checks if point uk is in range [0,1]
+		* if not, then uk is set to 0 resp. 1
+		* @param[in,out] uk
+		*/
+		void check_new_vals(T& uk); 
```

Comment:
Maybe rename? It is not really checking (which would probably return a `bool`) but setting.